src/datasources.js

```diff
@@ -300,7 +307,14 @@ define([
         }
         function getNodes(source, path, info) {
             var nodes = _.chain(source && source.structure)
-                .map(node(source, path, info))
+                .map(function (item, id) {
+                    if (item.merge) {
```

Comment:
The latter. `merge: true` was added because case properties and user properties have different base paths in the casedb (`/session/data` and `/session/context` respectively), but we don't want `data` and `context` to show up in the tree.